src/components/bulk-email-tool/bulk-email-form/BulkEmailForm.jsx

```diff
@@ -0,0 +1,221 @@
+                description="A warning shown to the user after submitting the email, to confirm the email recepients."
+              />
+            </p>
+            <ul>{selectedRecipients.map(group => <li key={group}>{group}</li>)}</ul>
+            <p>
+              <FormattedMessage
+                id="bulk.email.task.alert.warning"
+                defaultMessage="CAUTION! When you select Send Email, your email message is added to the queue for sending, and cannot be cancelled."
+                description="Warns the user in an alert that the email may not be immediately sent out to users."
+              />
+            </p>
+          </>
+        )}
+        close={event => {
+          if (event.target.name === 'continue') {
+            createEmailTask();
+          }
+          closeTaskAlert();
+        }}
+      />
+      <Form>
+        <BulkEmailRecepient
+          selectedGroups={selectedRecipients}
+          handleCheckboxes={onRecepientChange}
+          isValid={emailFormValidation.recepients}
+        />
+        <Form.Group controlId="emailSubject">
+          <Form.Label>
+            <FormattedMessage
+              id="bulk.email.subject.label"
+              defaultMessage="Subject:"
+              description="Email subject line input label. Meant to have colon or equivilant punctuation."
+            />
+          </Form.Label>
+          <Form.Control name="subject" className="w-lg-50" onChange={onSubjectChange} />
+          {!emailFormValidation.subject && (
+            <Form.Control.Feedback className="px-3" hasIcon type="invalid">
+              <FormattedMessage
+                id="bulk.email.form.subject.error"
+                defaultMessage="A subject is required"
+                description="An Error message located under the subject line. Visible only on failure."
+              />
+            </Form.Control.Feedback>
+          )}
+        </Form.Group>
+        <Form.Group controlId="emailBody">
+          <Form.Label>
+            <FormattedMessage
+              id="bulk.email.body.label"
+              defaultMessage="Body:"
+              description="Email Body label. Meant to have colon or equivilant punctuation."
+            />
+          </Form.Label>
+          <TextEditor onInit={onInit} />
+          {!emailFormValidation.body && (
+            <Form.Control.Feedback className="px-3" hasIcon type="invalid">
+              <FormattedMessage
+                id="bulk.email.form.body.error"
+                defaultMessage="The message cannot be blank"
+                description="An error message located under the body editor. Visible only on failure."
+              />
+            </Form.Control.Feedback>
+          )}
+        </Form.Group>
+        <Form.Group className="d-flex flex-row">
+          <StatefulButton
+            variant="primary"
+            type="submit"
+            onClick={event => {
+              event.preventDefault();
```

Comment:
Dumb question for my personal knowledge; what significance does the `event.preventDefault();` line do here? What is it preventing?

Reply:
clicking the submit button to a form causes a full page refresh. We don't want that, because that will prevent the alert and the completed states from showing feedback to the user.

Reply:
Thanks, that makes sense!